scripts/scripts.js

```diff
@@ -61,6 +81,260 @@ const { loadArea, loadDelayed, setConfig } = await import(`${miloLibs}/utils/uti
+      content.append(s);
+    }
+  });
+}
+
+// footer
+async function buildFooter(main) {
+  const footer = document.createElement('footer');
+  footer.classList.add('content');
+
+  const resp = await fetch(`/footer.plain.html`);
+  const html = await resp.text();
+  footer.innerHTML = html; 
+
+  main.append(footer);
+}
+
+// internal banner
+async function buildInternalBanner(block) {
+  const title = block.querySelector('.page-title');
+  
+  if (title) {
+    const banner = document.createElement('div');
+    banner.classList.add('section', 'internal-banner');
+    
+    const div = document.createElement('div');
+    div.innerHTML = '<span class="icon icon-info"></span>INTERNAL';
+    div.classList.add("banner");
+    banner.append(div);
+    title.insertAdjacentElement('afterend', banner);
+    decorateIcons(banner);
+
+    const text = document.createElement('div');
+    text.classList.add("content", "last-updated");
+    text.innerHTML = '&nbsp;';
+    banner.append(text);
+    
+    const index = await fetchIndex('query-index');
+    const found = index.data.find((entry) => entry.path.indexOf(window.location.pathname) > -1);
+    if (found) {
+      var dateFormat = new Date(parseInt(found.lastModified + '000', 10));
+      text.innerHTML = `Last updated on ${getMonthShortName((dateFormat.getMonth()))} ${dateFormat.getDate()}, ${dateFormat.getFullYear()}`; 
+    }
+  }
+}
+
+// tables
+function fixTableHeaders(main) {
+  var tables = main.querySelectorAll('table');
+  tables.forEach((t) => {
+    // remove empty lines
+    var lines = t.querySelectorAll('tbody tr');
+    lines.forEach((l) => {
+      const content = l.innerHTML.replaceAll('<td></td>', '').trim();
+      if (content === '') {
+          l.remove();
+      }
+    });
+
+    // fix header rowspan
+    const headerFirstRow = t.querySelector('thead tr:first-of-type');
+    const nHeaderTDs = headerFirstRow.querySelectorAll('th')?.length;
+    const bodyFirstRow = t.querySelector('tbody tr:first-of-type');
+    const nBodyTDs = bodyFirstRow.querySelectorAll('td')?.length;
+    if (nHeaderTDs === 2 && nBodyTDs === 3) {
+        const tHead = t.querySelector('thead');
+        if (tHead) {
+            tHead.appendChild(bodyFirstRow);
+            bodyFirstRow.querySelector('td:empty')?.remove();
+            tHead.querySelector('tr:first-of-type th:first-of-type')?.setAttribute('rowspan', 2);
+        }
+    }
+});
+}
+
+// "on this page" section
+function buildOnThisPageSection(main) {
+  const layout = document.querySelector('.layout-container');
+  if (!layout) {
+    return;
+  }
+
+  const headings = main.querySelectorAll('h1, h2, h3, h4, h5, h6');
+
+  const container = document.createElement('div');
+  container.classList.add('on-this-page');
+
+  const title = document.createElement('h5');
+  title.textContent = "On this page:";
+  container.append(title);
+
+  headings.forEach((h) => {
+    if (h.closest('.page-title') === null) {
+      console.log(h.textContent);
```

Comment:
remove unneeded console logs

Reply:
Good catch! Done in [b06a284](https://github.com/adobecom/helpx-internal/pull/2/commits/b06a2844a1a3a05b9b2f9098e1834daf1e4acce8)